### PR TITLE
add profile throughput benchmark

### PR DESCRIPTION
**Pull request by grimoire**

### CODE REVIEW

File: benchmark/profile_throughput.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    engine = Engine(model_path, model_name)
+    tokenizer = engine.tokenizer
+
+    requests = sample_requests(dataset, num_prompts, tokenizer)
+
+    elapsed_time = engine.process_request(requests, concurrency)
+    total_num_tokens = sum(prompt_len + output_len
+                           for _, prompt_len, output_len in requests)
+    print(f'Throughput: {len(requests) / elapsed_time:.2f} requests/s, '
```
Comment (wangruohui) [inline benchmark/profile_throughput.py:134]: Shall we also output throughput including only generated tokens, which is `sum(output_len) / elapsed_time`?

Comment (lvhan028) [reply benchmark/profile_throughput.py]: May update the benchmark in readme. And provide another benchmark about turbomind vs vllm on req/min

**Comment by lvhan028** [comment]
May add readme in `benchmark` dir, presenting method to get test data

File: benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+wget https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json
+```
+
+Profiling your model with `profile_throughput.py`
+
+```bash
+python profile_throughput.py \
+ ShareGPT_V3_unfiltered_cleaned_split.json \
+ /path/to/your/model \
+ ${ModelType} \
```
Comment (lvhan028) [inline benchmark/README.md:19]: may use ‘model_name’ . PR #144 will remove `model_name`

**Review by lvhan028 — APPROVED** [review]

**Review by wangruohui — APPROVED** [review]

**Pull request merged**